File: LCX/ValidEndPoints/dsl/examples/authenticated_trading.r
library(httr)
library(jsonlite)

"get_account_info" <- function() {
  url <- "https://exchange-api.lcx.com/api/account"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
"get_balances" <- function() {
  url <- "https://exchange-api.lcx.com/api/balances"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
"get_asset_balance" <- function(asset) {
  url <- "https://exchange-api.lcx.com/api/balance"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
"get_trading_fees" <- function() {
  url <- "https://exchange-api.lcx.com/api/fees"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
"get_open_orders" <- function(offset, pair) {
  url <- "https://exchange-api.lcx.com/api/open"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
"get_order_details" <- function(orderId) {
  url <- "https://exchange-api.lcx.com/api/order"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
"get_order_history" <- function(offset, pair, fromDate, toDate, side, orderStatus, orderType) {
  url <- "https://exchange-api.lcx.com/api/orderHistory"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
"get_user_trade_history" <- function(offset, pair, fromDate, toDate) {
  url <- "https://exchange-api.lcx.com/api/uHistory"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
# POST method not supported for r
# PUT method not supported for r
# DELETE method not supported for r
# DELETE method not supported for r
"check_balance_workflow" <- function() {
  url <- "https://exchange-api.lcx.com/api/balances"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
# Workflow: CheckBalanceWorkflow (sequence)

# None method not supported for r
"place_order_workflow" <- function(BTC/USDC) {
  url <- "https://exchange-api.lcx.com/api/ticker"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
# Workflow: PlaceOrderWorkflow (sequence)

# POST method not supported for r
"get_order_info" <- function(order_response) {
  url <- "https://exchange-api.lcx.com/api/order"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
# None method not supported for r
"manage_open_orders_workflow" <- function(0) {
  url <- "https://exchange-api.lcx.com/api/orderHistory"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
# Workflow: ManageOpenOrdersWorkflow (sequence)

"get_trade_history" <- function(0) {
  url <- "https://exchange-api.lcx.com/api/uHistory"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
# None method not supported for r
# PUT method not supported for r
# DELETE method not supported for r
# None method not supported for r
"account_monitoring_workflow" <- function() {
  url <- "https://exchange-api.lcx.com/api/balances"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
# Workflow: AccountMonitoringWorkflow (sequence)

"get_open_orders" <- function(0) {
  url <- "https://exchange-api.lcx.com/api/open"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
"get_fees" <- function() {
  url <- "https://exchange-api.lcx.com/api/fees"
  response <- httr::GET(url)
  return(httr::content(response, "text"))
}
# None method not supported for r

File: LCX/ValidEndPoints/dsl/examples/authenticated_trading.nim
import httpclient
import json
import hmac
import base64


proc get_account_info*(): JsonNode =
  ## GET /api/account
  let url = "https://exchange-api.lcx.com/api/account"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_balances*(): JsonNode =
  ## GET /api/balances
  let url = "https://exchange-api.lcx.com/api/balances"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_asset_balance*(asset): JsonNode =
  ## GET /api/balance
  let url = "https://exchange-api.lcx.com/api/balance"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_trading_fees*(): JsonNode =
  ## GET /api/fees
  let url = "https://exchange-api.lcx.com/api/fees"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_open_orders*(offset, pair): JsonNode =
  ## GET /api/open
  let url = "https://exchange-api.lcx.com/api/open"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_order_details*(orderId): JsonNode =
  ## GET /api/order
  let url = "https://exchange-api.lcx.com/api/order"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_order_history*(offset, pair, fromDate, toDate, side, orderStatus, orderType): JsonNode =
  ## GET /api/orderHistory
  let url = "https://exchange-api.lcx.com/api/orderHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_user_trade_history*(offset, pair, fromDate, toDate): JsonNode =
  ## GET /api/uHistory
  let url = "https://exchange-api.lcx.com/api/uHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

# POST method not supported for nim
# PUT method not supported for nim
# DELETE method not supported for nim
# DELETE method not supported for nim
proc check_balance_workflow*(): JsonNode =
  ## GET /api/balances
  let url = "https://exchange-api.lcx.com/api/balances"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

# Workflow: CheckBalanceWorkflow (sequence)

# None method not supported for nim
proc place_order_workflow*(BTC/USDC): JsonNode =
  ## GET /api/ticker
  let url = "https://exchange-api.lcx.com/api/ticker"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

# Workflow: PlaceOrderWorkflow (sequence)

# POST method not supported for nim
proc get_order_info*(order_response): JsonNode =
  ## GET /api/order
  let url = "https://exchange-api.lcx.com/api/order"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

# None method not supported for nim
proc manage_open_orders_workflow*(0): JsonNode =
  ## GET /api/orderHistory
  let url = "https://exchange-api.lcx.com/api/orderHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

# Workflow: ManageOpenOrdersWorkflow (sequence)

proc get_trade_history*(0): JsonNode =
  ## GET /api/uHistory
  let url = "https://exchange-api.lcx.com/api/uHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

# None method not supported for nim
# PUT method not supported for nim
# DELETE method not supported for nim
# None method not supported for nim
proc account_monitoring_workflow*(): JsonNode =
  ## GET /api/balances
  let url = "https://exchange-api.lcx.com/api/balances"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

# Workflow: AccountMonitoringWorkflow (sequence)

proc get_open_orders*(0): JsonNode =
  ## GET /api/open
  let url = "https://exchange-api.lcx.com/api/open"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_fees*(): JsonNode =
  ## GET /api/fees
  let url = "https://exchange-api.lcx.com/api/fees"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

# None method not supported for nim
